### ISSUE TITLE
train_supervised.py

### ISSUE THREAD

**Issue by AZ1229-forever** [romanlee6/langground#2]
Hello! I have a question and I hope you can answer it.
When I was running train_supervised.py, I tried two methods, “supervised_v0_half_minComm”，and“supervised_v0_exact_norm",
but the results seem to be quite different from those in your paper. May I ask which method should be used to obtain the "langground" shown in the following figure?
Here are the parameters I used during the runtime.
python ../../main.py --env_name predator_prey --nagents 3 --nprocesses 0 --num_epochs 500  --epoch_size 10 --hid_size 256 --lrate 0.001 --detach_gap 10 --supervised_gamma 1 --sampling_method ind --data_path ../../LLM/embedded_256_offline_llm_dataset_v0_27_1.csv --comm_dim 256 --max_steps 20 --vision 0 --dim 5 --exp_name supervised_v0_half_minComm --save_every 100 --supervised_comm --gate_reward_curriculum --min_comm_loss --eta_comm_loss 1. --ic3net --recurrent 

<img width="210" alt="Image" src="https://github.com/user-attachments/assets/1a6f6459-dc24-406f-b906-077398d46feb" />

Here are the graphs obtained from my training.

<img width="353" alt="Image" src="https://github.com/user-attachments/assets/2489ccd7-cb64-495c-ae75-d3546654810e" />

Thank you for your answer. Also, I wish you a happy new year!

**Comment by romanlee6** [comment]
"supervised_v0_exact_norm" should result in the same learning curve of "langground". You might want to train a bit longer for the policy to converge. For example try this one:

python ../../main.py --env_name predator_prey --nagents 3 --nprocesses 0 --num_epochs 2000  --epoch_size 10 --hid_size 256 --lrate 0.001 --detach_gap 10 --supervised_gamma 1 --sampling_method exact --data_path ../../LLM/embedded_256_offline_llm_dataset_v0_27_1.csv --comm_dim 256 --max_steps 20 --vision 0 --dim 5 --exp_name supervised_v0_exact_norm --save_every 100 --restore --norm_comm --supervised_comm --ic3net --recurrent --seed 1

**Comment by AZ1229-forever** [comment]
> "supervised_v0_exact_norm" should result in the same learning curve of "langground". You might want to train a bit longer for the policy to converge. For example try this one:
> 
> python ../../main.py --env_name predator_prey --nagents 3 --nprocesses 0 --num_epochs 2000 --epoch_size 10 --hid_size 256 --lrate 0.001 --detach_gap 10 --supervised_gamma 1 --sampling_method exact --data_path ../../LLM/embedded_256_offline_llm_dataset_v0_27_1.csv --comm_dim 256 --max_steps 20 --vision 0 --dim 5 --exp_name supervised_v0_exact_norm --save_every 100 --restore --norm_comm --supervised_comm --ic3net --recurrent --seed 1

Okay, I'll give it another try. Thank you for your answer.
